test/python/tensorflow/tf_unittest_runner.py

```diff
@@ -267,7 +268,13 @@ def read_tests_from_manifest(manifestfile, tensorflow_path):
     Reads a file that has include & exclude patterns,
     Returns a list of leaf-level single testcase, no duplicates
     """
-    list_of_tests = []
+
+    def invalidate_A_after_B(AList, BList):
+        for aTest in BList:
+            if aTest in AList:
+                AList.remove(aTest)
+
```

Comment:
Use "sets" for tests and you can simply take a set difference for this. You'll also be able to eliminate any duplicated tests as a result of using sets.